src/clients/cli.rs

```diff
@@ -189,6 +189,43 @@ impl Docker for Cli {
         info.network_settings.ports.into_ports()
     }
 
+    fn gateway(&self, id: &str) -> String {
```

Comment:
Two things regarding these functions:

1. I am not totally convinced we need them and would prefer if we keep the API surface of this traits as small as possible. Like I commented on the test, I believe using docker networks and default ports should make it possible to let the zookeeper and kafka image interact without having to expose this. That would also have the benefit that we provide a better out-of-the-box experience compared to what users currently have to do in a test to set things up properly.

2. IF it turns out that we still need these functions then we should at least reduce the duplication within the implementation. Currently, we have the code for calling `inspect` three times.